File: internal/src/main/java/dev/mvmo/sicklang/internal/object/ret/ReturnValueObject.java
package dev.mvmo.sicklang.internal.object.ret;

import dev.mvmo.sicklang.internal.object.ObjectType;
import dev.mvmo.sicklang.internal.object.SickObject;

public record ReturnValueObject(SickObject value) implements SickObject {

    @Override
    public String inspect() {
        return value.inspect();
    }

    @Override
    public ObjectType objectType() {
        return ObjectType.RETURN_VALUE;
    }
}

File: internal/src/main/java/dev/mvmo/sicklang/internal/object/ret/ReturnValueObject.kt
package dev.mvmo.sicklang.internal.`object`.ret

import dev.mvmo.sicklang.internal.`object`.ObjectType
import dev.mvmo.sicklang.internal.`object`.SickObject

class ReturnValueObject(val value: SickObject) : SickObject {

    override fun inspect() =
        value.inspect()

    override fun objectType() =
        ObjectType.RETURN_VALUE

}